vault/expiration.go

```diff
@@ -860,27 +860,19 @@ func (m *ExpirationManager) Stop() error {
 	close(m.quitCh)
 
 	m.pendingLock.Lock()
-	// Replacing the entire map would cause a race with
-	// a simultaneous WalkTokens, which doesn't hold pendingLock.
-	m.pending.Range(func(key, value interface{}) bool {
-		info := value.(pendingInfo)
-		info.timer.Stop()
-		m.pending.Delete(key)
-		return true
-	})
+	oldPending := m.pending
+	m.pending, m.nonexpiring, m.irrevocable = sync.Map{}, sync.Map{}, sync.Map{}
 	m.leaseCount = 0
-	m.nonexpiring.Range(func(key, value interface{}) bool {
-		m.nonexpiring.Delete(key)
-		return true
-	})
 	m.uniquePolicies = make(map[string][]string)
-	m.irrevocable.Range(func(key, _ interface{}) bool {
-		m.irrevocable.Delete(key)
-		return true
-	})
 	m.irrevocableLeaseCount = 0
 	m.pendingLock.Unlock()
 
+	go oldPending.Range(func(key, value interface{}) bool {
+		info := value.(pendingInfo)
+		info.timer.Stop()
```

Comment:
By letting this continue to run after Stop returns, for those leases which are about to expire and which we haven't yet stopped the timer, we expose ourselves to complications based on what the expireFunc is.  e.g. expireLeaseStrategyFairsharing grabs the read statelock.

Reply:
But if Stop is being called, that means we're in preSeal, and what happens next is postUnseal will wind up creating a new ExpirationManager.  So maybe we can just replace expireFunc with a no-op in this instance.  What about any that are in the process of already expiring?